### PR TITLE
Add check for vnet location for Azure Spring Cloud

### PR DESCRIPTION
**Pull request by gavinfish**
---

For VNet Injection instance, Azure Spring Cloud instance should be in the same location as the target VNet.

### General Guidelines

- [X] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [X] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Spring Cloud

File: src/spring-cloud/azext_spring_cloud/_validators.py [hunk 1]
```diff
@@ -180,6 +181,11 @@ def validate_vnet(cmd, namespace):
         raise CLIError('--app-subnet and --service-runtime-subnet should not be the same.')
 
     vnet_obj = _get_vnet(cmd, vnet_id)
+    instance_location = namespace.location
```
Comment (yungezz) [inline src/spring-cloud/azext_spring_cloud/_validators.py:184]: will there be format difference? like `west us` vs `westus`. :)

Comment (gavinfish) [reply src/spring-cloud/azext_spring_cloud/_validators.py:184]: Fixed.

**Review by yungezz — APPROVED** [review]

**Comment by yungezz** [comment]
pls resolve conflict, then good to go.

**Pull request merged**